src/SonarQube.Bootstrapper/Program.cs

```diff
@@ -56,20 +62,27 @@ public static int Execute(string[] args, ILogger logger)
                 return SuccessCode;
             }
 
-            if (!ArgumentProcessor.TryProcessArgs(args, logger, out IBootstrapperSettings settings))
+            try
             {
-                logger.ResumeOutput();
-                // The argument processor will have logged errors
-                Environment.ExitCode = ErrorCode;
-                return ErrorCode;
-            }
+                if (!ArgumentProcessor.TryProcessArgs(args, logger, out IBootstrapperSettings settings))
+                {
+                    logger.ResumeOutput();
+                    // The argument processor will have logged errors
+                    Environment.ExitCode = ErrorCode;
+                    return ErrorCode;
+                }
 
-            var processorFactory = new DefaultProcessorFactory(logger, GetLegacyTeamBuildFactory(),
-                GetCoverageReportConverter());
-            var bootstrapper = new BootstrapperClass(processorFactory, settings, logger);
-            var exitCode = bootstrapper.Execute();
-            Environment.ExitCode = exitCode;
-            return exitCode;
+                var processorFactory = new DefaultProcessorFactory(logger, GetLegacyTeamBuildFactory(),
+                    GetCoverageReportConverter());
+                var bootstrapper = new BootstrapperClass(processorFactory, settings, logger);
+                var exitCode = bootstrapper.Execute();
+                Environment.ExitCode = exitCode;
+                return exitCode;
+            }
```

Comment:
Should we catch exception and trace it? It would tell us which type failed to load.

Reply:
Why not create a global exception handler? E.g. use System.AppDomain.CurrentDomain.UnhandledException

Reply:
That shouldn't be necessary, should it? If it's an unhandled exception it should be logged to console anyway, so we should see the "failed to load assembly X or one of it dependencies" anyway.